src/confd/src/confd/ietf-system.c

```diff
@@ -579,10 +582,39 @@ static int is_valid_username(const char *user)
 	return 1;
 }
 
+static char *sys_find_usable_shell(sr_session_ctx_t *sess, char *name)
+{
+	char *shell = NULL;
+	char xpath[256];
+	sr_data_t *cfg;
+
+	snprintf(xpath, sizeof(xpath), XPATH_AUTH_"/user[name='%s']/infix-system:shell", name);
+	if (!sr_get_data(sess, xpath, 0, 0, 0, &cfg)) {
+		struct lyd_node *node;
+
+		if (!lyd_find_path(cfg->tree, xpath, 0, &node))
+			shell = (char *)lyd_get_value(node);
+	}
+
+	/* Verify the given shell exists (and is a login shell) */
+	if (!shell)
+		shell = LOGIN_SHELL;
+	else if (!whichp(shell))
+		shell = LOGIN_SHELL;
```

Comment:
Won't this still allow me to set `/usr/bin/mg`, for example, as my shell? Which is then disallowed by `/etc/shells` when I attempt to login?

Reply:
Yeah, you're right.  We should probably add something like https://github.com/kernelkit/infix/commit/76c09ba3bdf4d94a11b1bc5bcdd24c396f3d7863 to the model instead, and this passage could instead be reduced to falling back to `/bin/false`.

Reply:
We'll leave it as-is for now, and track this issue separately in #95.